include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,470 @@
+	BATTERY_PRESENCE_NO = 0,
+	BATTERY_PRESENCE_YES = 1,
+	BATTERY_PRESENCE_UNKNOWN,
+};
+
+enum battery_cutoff_states {
+	BATTERY_CUTOFF_STATE_NORMAL = 0,
+	BATTERY_CUTOFF_STATE_CUT_OFF,
+	BATTERY_CUTOFF_STATE_PENDING,
+};
+
+enum battery_disconnect_state {
+	BATTERY_DISCONNECTED = 0,
+	BATTERY_NOT_DISCONNECTED,
+	BATTERY_DISCONNECT_ERROR,
+};
+
+struct battery_static_info {
+	uint16_t design_capacity;
+	uint16_t design_voltage;
+	uint32_t cycle_count;
+	uint16_t serial[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	char manufacturer[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	char device_name[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	char chemistry[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	uint8_t vendor_param[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+};
+
+/* Battery parameters */
+struct batt_params {
+	int temperature; /* Temperature in 0.1 K */
+	int state_of_charge; /* State of charge (percent, 0-100) */
+	int voltage; /* Battery voltage (mV) */
+	int current; /* Battery current (mA); negative=discharging */
+	int desired_voltage; /* Charging voltage desired by battery (mV) */
+	int desired_current; /* Charging current desired by battery (mA) */
+	int remaining_capacity; /* Remaining capacity in mAh */
+	int full_capacity; /* Capacity in mAh (might change occasionally) */
+	int display_charge; /* Display charge in 10ths of a % (1000=100.0%) */
+	int status; /* Battery status */
+	enum battery_present is_present; /* Is the battery physically present */
+	int flags; /* Flags */
+};
+
+struct ship_mode_info {
+	/*
+	 * Write Block Support. If wb_support is true, then we use a i2c write
+	 * block command instead of a 16-bit write. The effective difference is
+	 * that the i2c transaction will prefix the length (2) when wb_support
+	 * is enabled.
+	 */
+	const uint8_t wb_support;
+	const uint8_t reg_addr;
+	const uint16_t reg_data[CONFIG_BATTERY_SHIP_MODE_WRITES];
+};
+
+/* Battery constants */
+struct battery_info {
+	/* Operation voltage in mV */
+	int voltage_max;
+	int voltage_normal;
+	int voltage_min;
+	/**
+	 * Pre-charge to fast charge threshold in mV,
+	 * default to voltage_min if not specified.
+	 * This option is only available on isl923x and rt946x.
+	 */
+	int precharge_voltage;
+	/* Pre-charge current in mA */
+	int precharge_current;
+	/* Working temperature ranges in degrees C */
+	int8_t start_charging_min_c;
+	int8_t start_charging_max_c;
+	int8_t charging_min_c;
+	int8_t charging_max_c;
+	int8_t discharging_min_c;
+	int8_t discharging_max_c;
+	uint8_t vendor_param_start;
+};
+
+/*
+ * Provide a 1 minute average of the current and voltage on the battery.
+ * Does not check for flags or whether those values are bad readings.
+ * See driver/battery/[your_driver].h/c for details on implementation and
+ * how the average is calculated.
+ */
+typedef int (*battery_get_avg_current)(const struct device *dev); /* in mA */
+typedef int (*battery_get_avg_voltage)(const struct device *dev); /* in mV */
```

Comment:
docblock for ...voltage, or the get_avg_voltage function will not be documented at all. Include @return to document units of return value, not a comment!